social_network/posts.py

```diff
@@ -1,33 +1,47 @@
 from datetime import datetime
 
 
-# class Post(object):
-#     def __init__(self, text, timestamp=None):
-#         pass
-#
-#     def set_user(self, user):
-#         pass
-
-
-# class TextPost(...):  # Inherit properly
-#     def __init__(self, text, timestamp=None):
-#         pass
-#
-#     def __str__(self):
-#         pass
-
-
-# class PicturePost(...):  # Inherit properly
-#     def __init__(self, text, image_url, timestamp=None):
-#         pass
-#
-#     def __str__(self):
-#         pass
-
-
-# class CheckInPost(...):  # Inherit properly
-#     def __init__(self, text, latitude, longitude, timestamp=None):
-#         pass
-#
-#     def __str__(self):
-#         pass
+class Post(object):
+    user = None
+    
+    def __init__(self, text, timestamp=None):
+        self.text = text
+        
+
+    def set_user(self, user):
+        self.user = user
+
+    def __repr__(self):
+        rep = "{}'s post at {}".format(self.user, self.timestamp.strftime('%A, %b %d, %Y'))
+        return rep
+    
+class TextPost(Post):  # Inherit properly
+    def __init__(self, text, timestamp=None):
+        self.text = text
+        self.timestamp = timestamp
+
+    def __str__(self):
+        return '@{}: "{}"\n\t{}'.format(self.user, self.text, self.timestamp.strftime('%A, %b %d, %Y'))
```

Comment:
My preference would be to define timestamp's formatting in init so that I do not have to type the same thing 3x, but this obviously works as well :)

Nice work!